### PR TITLE
Added support for union types for components

### PR DESCRIPTION
**Pull request by lulco**

### CODE REVIEW

File: src/LatteContext/Collector/ComponentCollector.php [hunk 1]
```diff
@@ -164,14 +165,31 @@ private function findAssignToThis(Assign $node, Scope $scope, ClassReflection $c
...
-        $components = [];
-
-        // TODO support union types
+        $componentType = null;
         if ($componentArgType instanceof ObjectType && $componentArgType->isInstanceOf('Nette\ComponentModel\IComponent')->yes()) {
+            $componentType = $componentArgType;
+        } elseif ($componentArgType instanceof UnionType) {
+            $componentTypes = [];
+            foreach ($componentArgType->getTypes() as $type) {
+                if ($type instanceof ObjectType && $type->isInstanceOf('Nette\ComponentModel\IComponent')->yes()) {
```
Comment (lulco) [inline src/LatteContext/Collector/ComponentCollector.php:182]: not sure if all types from union have to by IComponent or some of them :)

File: src/LatteContext/Collector/ComponentCollector.php [hunk 2]
```diff
@@ -181,12 +199,7 @@ private function buildComponents(Node $node, Scope $scope, ClassReflection $clas
             }
         }
 
-        $lattePhpDoc = $this->lattePhpDocResolver->resolveForNode($node, $scope);
```
Comment (lulco) [inline src/LatteContext/Collector/ComponentCollector.php:184]: moved to method start

File: src/LatteContext/Collector/ComponentCollector.php [hunk 3]
```diff
@@ -181,12 +199,7 @@ private function buildComponents(Node $node, Scope $scope, ClassReflection $clas
...
-        if ($lattePhpDoc->isIgnored()) {
-            return null;
-        }
         if ($lattePhpDoc->hasComponents()) {
-            $components = [];
```
Comment (lulco) [inline src/LatteContext/Collector/ComponentCollector.php:189]: other thing I'm not sure with... If I collect some components, and there are some in phpdoc I think I don't want to reset all collected. Just overwrite those which are in php doc and add missing from php doc

Comment (MartinMystikJonas) [reply src/LatteContext/Collector/ComponentCollector.php:189]: Correct we did same for variables

File: src/LatteContext/Collector/ComponentCollector.php (hunk 1 above)
Comment (MartinMystikJonas) [reply src/LatteContext/Collector/ComponentCollector.php:182]: I would take what is component and ignore others

Comment (lulco) [reply src/LatteContext/Collector/ComponentCollector.php:182]: ok, than it is implemented this way

**Review by MartinMystikJonas — APPROVED** [review]

**Pull request merged**